include/hpp/fcl/collision_object.h

```diff
@@ -93,14 +94,14 @@ class HPP_FCL_DLLAPI CollisionGeometry
     && aabb_center == other.aabb_center
     && aabb_radius == other.aabb_radius
     && aabb_local == other.aabb_local
-//    && user_data == other.user_data
+    && isEqual(other)
     ;
   }
 
   /// \brief Difference operator
   bool operator!=(const CollisionGeometry & other) const
   {
-    return !(*this == other);
+    return typeid(*this) == typeid(other) && isNotEqual(other);
```

Comment:
I guess now it should be
```suggestion
    return isNotEqual(other);
```

Reply:
Thanks